rs/bitcoin/ckbtc/minter/src/main.rs

```diff
@@ -92,6 +92,14 @@ async fn distribute_kyt_fee() {
 #[cfg(feature = "self_check")]
 #[update]
 async fn refresh_fee_percentiles() {
+    // Use `TimerLogicGuard` here because:
+    // 1. `estimiate_fee_per_vbyte` could potentially change the state.
+    // 2. `estimiate_fee_per_vbyte` is also called from timer
+    //    `TaskType::ProcessLogic` and `TaskType::RefreshFeePercentiles`.
+    let _guard = match crate::guard::TimerLogicGuard::new() {
+        Some(guard) => guard,
+        None => return,
+    };
```

Comment:
I think you also need to guard the code block `TaskType::RefreshFeePercentiles` [here](https://github.com/dfinity/ic/blob/fe0aacc284624e953d881db9d51936de1cd2be7a/rs/bitcoin/ckbtc/minter/src/lib.rs#L1201), no?

Reply:
Yes indeed. Good catch! Just added it.